src/SelfUpdateCommand.php

```diff
@@ -0,0 +1,162 @@
+ *
+ * @author Alexander Menk <alex.menk@gmail.com>
+ */
+class SelfUpdateCommand extends Command
+{
+    private $command;
+
+    protected $gitHubRepository;
+
+    protected $currentVersion;
+
+    public function __construct($name = null, $currentVersion = null, $gitHubRepository = null)
+    {
+        parent::__construct($name);
+        $this->currentVersion = $currentVersion;
+        $this->gitHubRepository = $gitHubRepository;
+    }
+
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function configure()
+    {
+        $this
+            ->setName('self-update')
+            ->setAliases(array( 'selfupdate' ))
+            ->setDescription('Updates the robo.phar to the latest version.')
+            ->setHelp(
+                <<<EOT
+The <info>self-update</info> command checks github for newer
+versions of robo and if found, installs the latest.
+EOT
+            );
+    }
+
+    protected function getLatestReleaseFromGithub()
+    {
+        $opts = [
+            'http' => [
+                'method' => 'GET',
+                'header' => [
+                    'User-Agent: ' . Robo::APPLICATION_NAME . ' Self-Update (PHP)'
+                ]
+            ]
+        ];
+
+        $context = stream_context_create($opts);
+
+        $releases = file_get_contents('https://api.github.com/repos/' . $this->gitHubRepository . '/releases', false, $context);
+        $releases = json_decode($releases);
+
+        if (! isset($releases[0])) {
+            throw new \Exception('API error - no release found at GitHub repository ' . $this->gitHubRepository);
+        }
+
+        $version = $releases[0]->tag_name;
+        $url     = $releases[0]->assets[0]->browser_download_url;
+
+        return [ $version, $url ];
+    }
+
+    public function isEnabled()
+    {
+        return !empty(\Phar::running());
+    }
+
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+
+        $localFilename = realpath($_SERVER['argv'][0]) ?: $_SERVER['argv'][0];
+        $programName   = basename($localFilename);
+        $tempFilename  = dirname($localFilename) . '/' . basename($localFilename, '.phar') . '-temp.phar';
+
+        // check for permissions in local filesystem before start connection process
+        if (! is_writable($tempDirectory = dirname($tempFilename))) {
+            throw new \Exception(
+                $programName . ' update failed: the "' . $tempDirectory .
+                '" directory used to download the temp file could not be written'
+            );
+        }
+
+        if (! is_writable($localFilename)) {
+            throw new \Exception(
+                $programName . ' update failed: the "' . $localFilename . '" file could not be written (execute with sudo)'
+            );
+        }
+
+        list( $latest, $downloadUrl ) = $this->getLatestReleaseFromGithub();
+
+
+        if ($this->currentVersion == $latest) {
```

Comment:
Did you decide that handling dev versions was unnecessary since this only works for phars? That's probably sufficient. An explanatory comment here would be helpful, though.

Reply:
Yes, I thought it's fine because of  the non-phar-disabling.